src/Core/Context/LegacyController.php

```diff
@@ -0,0 +1,281 @@
+     */
+    public array $errors = [];
+
+    public string $path;
+
+    public string $currentIndex;
+
+    public string $content;
+
+    public array $warnings = [];
+
+    public array $informations = [];
+
+    public array $confirmations = [];
+
+    public string|false $shopShareDatas = false;
+
+    public array $_languages = [];
+
+    public int $default_form_language;
+
+    public int $allow_employee_form_lang;
+
+    public string $layout = 'layout.tpl';
+
+    public bool $bootstrap = false;
+
+    public string $template = 'content.tpl';
+
+    public string $table = 'configuration';
+
+    public ?string $list_id;
+
+    public array $tabAccess;
+
+    public int $id = -1;
+
+    public bool $required_database = false;
+
+    // Security token
+    public ?string $token;
+
+    // "shop" or "group_shop"
+    public ?string $shopLinkType;
+
+    public array $tpl_option_vars = [];
+
+    public array $tpl_view_vars = [];
+
+    public array $tpl_required_fields_vars = [];
+
+    public ?string $base_tpl_view = null;
+
+    public ?string $base_tpl_form = null;
+
+    // If you want more fieldsets in the form
+    public bool $multiple_fieldsets = false;
+
+    public array $fields_value = [];
+
+    public ?int $max_image_size = null;
+
+    // Automatically join language table if true
+    public bool $lang = false;
+
+    // Cache for translations
+    public array $cache_lang = [];
+
+    // Required_fields to display in the Required Fields form
+    public array $required_fields = [];
+
+    public string $override_folder;
+
+    // Name and directory where class image are located
+    public array $fieldImageSettings = [];
+
+    // Image type
+    public string $imageType = 'jpg';
+
+    // Current controller name without suffix
+    public string $controller_name;
+
+    public int $multishop_context = -1;
+
+    public bool $multishop_context_group = true;
+
+    // Bootstrap variable
+    public bool $show_page_header_toolbar = false;
+
+    // Bootstrap variable
+    public string $page_header_toolbar_title;
+
+    // Bootstrap variable
+    public array|Traversable $page_header_toolbar_btn = [];
+
+    // Bootstrap variable
+    public ?bool $show_form_cancel_button;
+
+    public ?string $admin_webpath;
+
+    /** @var array<string, string|array> */
+    public array $modals = [];
+
+    // Auth cookie lifetime
+    public const AUTH_COOKIE_LIFETIME = 3600;
+
+    public array $_conf;
+
+    public array $_error;
+
+    public int $_lang;
+
+    public string $tpl_folder;
+
+    public function __construct(
+        ContainerInterface $container,
+        string $controller_name,
+        string $controller_type,
+        string $php_self,
+        int $multishop_context,
+        ?string $className,
+        int $id,
+        ?string $token,
+        array $_conf,
+        array $_error,
+        ?string $shopLinkType,
+        string $override_folder,
+        string $tpl_folder,
+        ?string $admin_webpath,
```

Comment:
Following my previous comments about what should be kept and what could be cleaned, these variables for example are clearly still useful since we populate them appropriately

The CSS files/ js files are also meaningful since we need them in the layout, the warnings/errors/notifications I'm not sure any more but I believe, if it's not already done, we did plan on handling a backward compatibility so that they are displayed in the symfony layout